### ISSUE TITLE
Question: Get package name of consumer?

### ISSUE THREAD

**Issue by jimlyas** [google/ksp#1035]
So I know we can generate new file using `codeGenerator` from `SymbolProcessorEnvironment`.

We can also required to define package name for the file. 

My question is can we set the package name to the package name of consumer?

**Comment by jimlyas** [comment]
Thanks for the reply, let me rephrase my question.

I want my generated code have the same **root package name** as the consumer. I knew how to get package name from annotated file, and it could be child package name and not the root package.


Can we achieve this?

**Comment by neetopia** [comment]
Sorry I didn't get it, can you express your requirement in code snippet?

**Comment by jimlyas** [comment]
Okay, let's say my annotation processor have a consumer with a package name like this:

```
com.sample.application
├── nested
│   └── AnnotatedFile.kt
├── something
│   └── AnotherAnnotatedFile.kt
└── somethingelse
```

I can get the package name of `AnnotatedFile.kt` and `AnotherAnnotatedFile.kt` to get result like 
`com.sample.application.nested` and `com.sample.application.something`.


and i can put it in here when generating new file:

```kotlin
environment.codeGenerator.createNewFile(
    Dependencies(false, *list.mapNotNull { it.containingFile }.toList().toTypedArray()),
    "com.sample.application.nested",
    "SampleGeneratedFile"
)
```


But what I really want is to have the generated file have the same root package name as the consumer, so `com.sample.application` only.

Does KSP can read the root package name of the consumer that used it?

**Comment by neetopia** [comment]
closing as inactive, feel free to reopen if any updates.
